yt_dlp/extractor/cda.py

```diff
@@ -351,3 +353,52 @@ def extract_format(page, version):
             extract_format(webpage, resolution)
 
         return merge_dicts(info_dict, info)
+
+
+class CDAFolderIE(InfoExtractor):
+    _MAX_PAGE_SIZE = 36
+    _VALID_URL = r'https?://(?:www\.)?cda\.pl/(?P<channel>\w+)/folder/(?P<id>\d+)'
+    _TESTS = [
+        {
+            'url': 'https://www.cda.pl/domino264/folder/31188385',
+            'info_dict': {
+                'id': '31188385',
+                'title': 'SERIA DRUGA',
+            },
+            'playlist_count': 13,
+        },
+        {
+            'url': 'https://www.cda.pl/smiechawaTV/folder/2664592/vfilm',
+            'info_dict': {
+                'id': '2664592',
+                'title': 'VideoDowcipy - wszystkie odcinki',
+            },
+            'playlist_count': 71,
+        },
+        {
+            'url': 'https://www.cda.pl/DeliciousBeauty/folder/19129979/vfilm',
+            'info_dict': {
+                'id': '19129979',
+                'title': 'TESTY KOSMETYKÓW',
+            },
+            'playlist_count': 139,
+        }]
+
+    def _real_extract(self, url):
+        folder_id, channel = self._match_valid_url(url).group('id', 'channel')
+
+        webpage = self._download_webpage(url, folder_id)
+
+        def extract_page_entries(channel_name, folder_id, page):
+            webpage = self._download_webpage(
+                f'https://www.cda.pl/{channel_name}/folder/{folder_id}/vfilm/{page + 1}', folder_id,
+                f'Downloading page {page + 1}', expected_status=404)
+            items = re.findall(r'<a[^>]+href="/video/([0-9a-z]+)" class=', webpage)
+            for video_id in items:
+                yield self.url_result(f'https://www.cda.pl/video/{video_id}', CDAIE, video_id)
+
+        entries = OnDemandPagedList(
+            functools.partial(extract_page_entries, channel, folder_id),
+            self._MAX_PAGE_SIZE)
+
```

Comment:
The variables can be captured so we can use them directly instead. That way we also do not need `functools` anymore:
```suggestion
        def extract_page_entries(page):
            webpage = self._download_webpage(
                f'https://www.cda.pl/{channel}/folder/{folder_id}/vfilm/{page + 1}', folder_id,
                f'Downloading page {page + 1}', expected_status=404)
            items = re.findall(r'<a[^>]+href="/video/([0-9a-z]+)" class=', webpage)
            for video_id in items:
                yield self.url_result(f'https://www.cda.pl/video/{video_id}', CDAIE, video_id)

        entries = OnDemandPagedList(extract_page_entries, self._MAX_PAGE_SIZE)
```

Reply:
I'd increment `page` by 1 at the beginning of the function so you don't need to `+ 1` in multiple places

---

Why is the ` class='` part necessary in the pattern below?
> ```py
>             items = re.findall(r'<a[^>]+href="/video/([0-9a-z]+)" class=', webpage)
> ```


would this work instead?
```
             items = re.findall(r'<a [^>]*\bhref="/video/([\da-z]+)"', webpage)
```

---

Also, if applying Grub's above suggestion then I would also inline the `OnDemandPagedList` in the return line below:

```py
       return self.playlist_result(
           OnDemandPagedList(extract_page_entries, self._MAX_PAGE_SIZE),
           folder_id, self._og_search_title(webpage))
```